plugins/catalog-import/src/api/CatalogImportApi.ts

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright 2020 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { createApiRef } from '@backstage/core';
+import { PartialEntity } from '../util/types';
+
+export const catalogImportApiRef = createApiRef<CatalogImportApi>({
+  id: 'plugin.catalogimport.service',
+  description: 'Used by the catalog import plugin to make requests',
+});
+
+export interface CatalogImportApi {
+  submitPrToRepo(options: {
+    oAuthToken: string;
```

Comment:
Better have this happen internally in the implementation instead by depending on the appropriate auth APIs. This is hardcoded for GitHub anyway so might as well depend on GitHub auth directly. Having that work a bit more dynamically based on what type of repo we're working with is something we need to figure out in general anyway.

Reply:
How to use GitHub auth without the `useApi` hook? Can some GithubApi client be injected to the constructor of my `CatalogImportApi`?

Reply:
Well, I've done this exactly :)